trin-core/src/portalnet/types/metric.rs

```diff
@@ -45,6 +96,41 @@ mod test {
         }
     }
 
+    #[test]
+    fn distance_log2() {
+        fn prop(x: DhtPoint) -> TestResult {
+            let x = U256::from_big_endian(&x.0);
+            let distance = Distance(x);
+            let log2_distance = distance.log2();
+
+            match log2_distance {
+                Some(log2) => {
+                    let x_floor = U256::from(1u8) << (log2 - 1);
+
+                    if log2 == 256 {
+                        TestResult::from_bool(x >= x_floor)
+                    } else {
+                        let x_ceil = U256::from(1u8) << log2;
+                        TestResult::from_bool(x >= x_floor && x < x_ceil)
+                    }
+                }
+                None => TestResult::from_bool(distance.0.is_zero()),
+            }
+        }
+        quickcheck(prop as fn(DhtPoint) -> TestResult);
```

Comment:
In a quick read of `quickcheck` I couldn't find an obvious way to force it to test any particular examples. Normally I would want to see that a few points near the edge cases of exactly the log2 value as distance +/- 1 all pass.

If you know how to add that, that would be cool. Maybe another alternative is to manually run `prop()` with hard-coded arguments and match or unwrap the `TestResult`.

Not a big deal I think, but if it's quick, it would be a great addition.